src/charm.py

```diff
@@ -280,6 +280,7 @@ def _on_peer_relation_changed(self, event: RelationChangedEvent) -> None:
             unit_label
         ):
             self.unit_peer_data["unit-initialized"] = "True"
+            self.unit_peer_data["member-state"] = "online"
```

Comment:
just Q: why do we need to put the state into relation if we can always ask mysql itself?

Reply:
the storage of the member state in the peer relation was introduced when handling full cluster crash recovery (to skip some action like `update-status` where the charm could not access mysqld). it was then expanded to be used when the unit is waiting to start up and waiting to be restored.

we will be looking closely at this in a separate ticket ([DPE-1509](https://warthogs.atlassian.net/browse/DPE-1509)) to ensure that we are not storing state in the peer relation if we can avoid it.